frontend/src/scenes/retention/RetentionTable.tsx

```diff
@@ -29,8 +29,13 @@ export function RetentionTable({ inSharedMode = false }: { inSharedMode?: boolea
         retentionMeans,
         breakdownDisplayNames,
     } = useValues(retentionTableLogic(insightProps))
-    const { toggleBreakdown } = useActions(retentionTableLogic(insightProps))
+    const { toggleBreakdown, setSelectedInterval, setHoveredColumn } = useActions(retentionTableLogic(insightProps))
+    const { hoveredColumn } = useValues(retentionTableLogic(insightProps))
     const { openModal } = useActions(retentionModalLogic(insightProps))
+
+    const selectedInterval = retentionFilter?.selectedInterval ?? null
+    const allowSelectingColumns = !insightProps.dashboardId
```

Comment:
This should also be disabled for shared insights/dashboards and insights in notebooks.